### PR TITLE
Fix threshold_multiotsu regression

### LINKED ISSUE #4173: Regression in threshold_multiotsu
## Description

During my exploration for fixing #3980 I observed that `threshold_multiotsu` gives a different result on the `astronaut` image thanks to @sciunto https://github.com/scikit-image/scikit-image/issues/3980#issuecomment-521624136_

In fact, after refactoring,  the new version gives `0.45507812` while the original version gives `0.439453125`.

I also noticed that `threshold_multiotsu` does not warn the user that the method doesn't handle color images and do not raise error if the image has only one value as `threshold_otsu` does!

## Way to reproduce
```python
In [1]: from skimage.filters import threshold_multiotsu, threshold_otsu                                 

In [2]: from skimage.color import rgb2gray                                                              

In [3]: from skimage.data import moon, astronaut                                                        

In [4]: img = rgb2gray(astronaut())                                                                     

In [5]: threshold_multiotsu(img, 2)                                                                     
Out[5]: array([0.45507812])

```

## Version information

```                                                                               
3.7.3 (default, Mar 27 2019, 22:11:17) 
[GCC 7.3.0]
Linux-4.15.0-60-generic-x86_64-with-debian-buster-sid
scikit-image version: 0.16.dev0
numpy version: 1.16.4
```

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

Closes #4173 

I added a test for checking `threshold_multiotsu` result on `astronaut` image.

## Checklist

- [x] Unit tests
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/filters/tests/test_thresholding.py [hunk 1]
```diff
@@ -563,6 +564,9 @@ def test_multiotsu_output():
     thresholds = [64, 128]
     assert np.array_equal(thresholds, threshold_multiotsu(image))
 
+    image = color.rgb2gray(data.astronaut())
+    assert 0.438 < threshold_multiotsu(image, 2) < 0.440
```
Comment (sciunto) [inline skimage/filters/tests/test_thresholding.py]: Can you try to use assert_almost_equal please?

Comment (rfezzani) [reply skimage/filters/tests/test_thresholding.py]: Thank you @sciunto for your review. I will come soon with a new PR with a complete rewrite of `threshold_multiotsu` fixing the different issues. It is then may be not necessary to merge this PR.

Comment (sciunto) [reply skimage/filters/tests/test_thresholding.py]: We are on the way to release 0.16. To this end, a quick fix might be better than a complete rewrite that will take time to review. ;) Cc @jni.

**Comment by sciunto** [comment]
Great, thanks @rfezzani

**Pull request merged**